front/app/hooks/files/useSyncPhaseFiles.ts

```diff
@@ -23,20 +25,45 @@ export function useSyncPhaseFiles() {
       filesToRemove,
       fileOrdering,
     }: SyncPhaseFilesArguments) => {
-      // Get any files that we need to add
-      const filesToAddPromises = phaseFiles
-        .filter((file) => !file.remote)
-        .map((file) =>
-          addPhaseFile({
-            phaseId,
-            base64: file.base64 || '',
-            ordering: file.ordering!, // Use the ordering from drag-and-drop
-            name: file.name,
-            invalidate: false, // Prevents re-fetching the list after each update. We handle it once instead at the end.
+      // First, create any File Attachments for existing files from File Library
+      const filesToAttach = phaseFiles.filter(
+        (file) => !file.remote && file.id // Files with IDs that aren't remote yet are considered new attachments.
+      );
+      if (filesToAttach.length > 0) {
+        const filesToAttachPromises = filesToAttach.map((file) =>
+          addPhaseFileAttachment({
+            file_id: file.id || '',
+            attachable_id: phaseId,
+            attachable_type: 'Phase',
+          }).then((attachment) => {
+            // After attaching, also update the file's ordering
+            return updatePhaseFile({
+              phaseId,
+              fileId: attachment.data.id,
```

Comment:
This I found a bit strange - the BE seems to require the File Attachment ID here, not the File ID?

Reply:
Also I originally tried passing the ordering on the addPhaseFileAttachment directly, but that didn't seem to work?